### ISSUE TITLE
[Performance Regression] GPU decode speed drops to ~1/3 starting v0.13.0 (for GPU-embedding models)

### ISSUE THREAD

**Issue by pionio2** [google-ai-edge/LiteRT-LM#2472]
**My environment**

- Model: qwen3-0.6b
- Conversion/Bundling: quantized & bundled with litert-torch-nightly into a .litertlm file
- Quantization: dynamic 8-bit
- Platform: Android (Kotlin)
- Device: Samsung Galaxy S26 Ultra
- Backend: GPU
- Versions compared: v0.12.0 → v0.13.0

**Symptom**

- After upgrading from v0.12.0 to v0.13.0, the GPU backend inference (decode) speed dropped to roughly 1/3.
- Reproducible by changing only the version with the same model/device/settings.
- I'd like to confirm whether a code change between the two versions caused this regression, and whether it has been fixed in a later release (v0.13.1).

**Root Cause Analysis**

After diffing v0.12.0..v0.13.0, the culprit appears to be commit 7510ce74 ("Internal changes only.", PiperOrigin-RevId: 921824719).
It added the following condition to InitializeSampler() in runtime/executor/llm_litert_compiled_model_executor.cc:

```
// Disable GPU token copy for models that run embedding on the GPU.
const bool runs_embedding_on_gpu = (embedding_lookup_ == nullptr);   // newly added

sampler_handles_input_ =
    (!executor_settings_.GetAdvancedSettings().has_value() ||
     executor_settings_.GetAdvancedSettings()->sampler_handles_input) &&
    sampler_->CanHandleInput() && !signatures_.input_tokens.empty() &&
    !runs_embedding_on_gpu;   // newly added condition
```

When qwen3-0.6b runs on GPU, the embedding is performed inside the GPU graph, so embedding_lookup_ == nullptr. As a result, sampler_handles_input_, which was true in v0.12.0, is now forced to false starting in v0.13.0.

The value of sampler_handles_input_ drastically changes the per-token decode cost (DecodeInternal()):

> true (v0.12.0, fast): The sampler is wired through BindTensorsAndRunDecodeStatic, so as soon as a token is sampled it is bound on-device to the next step's input tensors and the next decode is chained — an on-device fused pipeline where the token never returns to the host.

> false (v0.13.0/v0.13.1, slow): Every token now requires ① decode → ② sample logits → ③ read the token id back to CPU → ④ re-upload it into the GPU input buffer via FillInputBufferWithToken → ⑤ next decode. This forces a CPU↔GPU sync and round-trip copy per token, stalling the GPU pipeline on every step.

For small models like 0.6b, per-step compute is short, so this round-trip overhead dominates — which explains the dramatic ~1/3 throughput observed.

**Questions**

1. What was the intent of disabling sampler_handles_input_ for GPU-embedding models in 7510ce74 (e.g., avoiding a correctness issue)?
2. Is the resulting decode slowdown an intended trade-off, or a regression?
3. Are there plans to restore the fused-input path for GPU-embedding models?
